Source/UnitEx.cpp

```diff
@@ -1129,14 +1138,19 @@ void UnitEx::SetAnimation(int animIdx, int animDelay, int animStartDelay, int bo
 	func(&(*unitArray)[unitID], 0, animIdx, animDelay, animStartDelay, boolInvisible, boolSkipDoDeath);
 }
 
+// HFL initialization must be checked before calling GetBeaconData
+BeaconData& GetBeaconData(int unitID)
+{
+	return *(BeaconData*)(reinterpret_cast<std::uintptr_t>(*unitArray) + (unitID * sizeof(OP2Unit)) + beaconDataOffset);

```

Comment:
Do we want the portion `(unitID * sizeof(OP2Unit))` to have a reinterpret cast uintptr_t as well?

Reply:
I would say no. The type of `sizeof` is `std::size_t`. It should make sense to add `std::uintptr_t` and `std::size_t`, with the result being a `std::uintptr_t`. That's like adding a `Point` with a `Vector` to get a new `Point`.